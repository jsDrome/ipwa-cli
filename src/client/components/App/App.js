import React from "react";
import { StyleSheet, View } from "react-native";

function App() {
  return (
    <View style={styles.app}>
    </View>
  );
}

const styles = StyleSheet.create({
  app: {
    height: '100vh',
    backgroundColor: '#1a73e8',
  },
});

export default App;
