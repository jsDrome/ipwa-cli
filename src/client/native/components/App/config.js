module.exports = {
  showBannerAd: false,
  showInterstitialAd: false,
  ad: {
    test: {
      bannerAdUnitId: 'ca-app-pub-3940256099942544/6300978111',
      interstitialAdUnitID: 'ca-app-pub-3940256099942544/4411468910',
    },
    production: {
      bannerAdUnitId: 'ca-app-pub-3940256099942544/6300978111',
      interstitialAdUnitID: 'ca-app-pub-3940256099942544/4411468910',
    },
  },
  webview: {
    userAgent: "Mozilla/5.0 (Linux; Android 7.0; SM-G930V Build/NRD90M) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.125 Mobile Safari/537.36",
    uri: "https://ipwa-app.web.app?app=true",
  },
};
