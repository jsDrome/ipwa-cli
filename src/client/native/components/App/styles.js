export default {
  app: {
    flex: 1,
    backgroundColor: '#000000',
  },
  webView: {
    height: '100%',
    marginTop: 30,
    backgroundColor: '#000000',
  },
  loader: {
    height: '100%',
    backgroundColor: '#000000',
  },
  bannerAd: {
    position: "absolute",
    bottom: 0,
  },
};
