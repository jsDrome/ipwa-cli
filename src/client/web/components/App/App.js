import React, { Fragment, useEffect } from "react";
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Meta from './Meta';
import Alert from 'Components/Alert/Alert';

import { checkForMessage } from 'Store/app/appActions';

const App = ({ children, checkForMessage, alert }) => {

  useEffect(checkForMessage, []);

  return <Fragment>
    <Meta />
    {alert && <Alert type={alert.type} duration={alert.duration}>{alert.message}</Alert>}
    {children}
  </Fragment>;
}

App.propTypes = {
  children: PropTypes.node,
};

App.defaultProps = {
  children: null,
};

const mapStateToProps = state => ({
  alert: alertSelector(state),
});

const alertSelector = state => state.app.alert;

const mapDispatchToProps = dispatch => ({
  checkForMessage: data => checkForMessage(data)(dispatch),
})

export {
  App,
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
