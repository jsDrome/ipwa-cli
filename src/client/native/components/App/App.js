import React from 'react';
import { View, StatusBar, ImageBackground } from 'react-native';
import { WebView } from 'react-native-webview';
import { AdMobBanner } from 'expo-ads-admob';

import styles from './styles';

const { showBannerAd, ad, webview: { uri, userAgent }, splashUri } = require('./config');

// eslint-disable-next-line no-undef
const env = __DEV__ ? "test" : "production"
const { bannerAdUnitId } = ad[env];

const App = () => <View style={styles.app}>
  <StatusBar barStyle="light-content" />
  <WebView
    userAgent={userAgent}
    javaScriptEnabled={true}
    domStorageEnabled={true}
    startInLoadingState={true}
    scrollEnabled={true}
    renderLoading={Loader}
    bounce={false}
    useWebKit={true}
    source={{ uri }}
    style={styles.webView} />
  {showBannerAd && <BannerAd />}
</View>;

const BannerAd = () => AdMobBanner && <AdMobBanner
  style={styles.bannerAd}
  bannerSize="fullBanner"
  adUnitID={bannerAdUnitId}
  testDeviceID="EMULATOR"
  servePersonalizedAds
  onDidFailToReceiveAdWithError={() => console.log('error from ad')} />;

const Loader = () => <ImageBackground source={{ uri: splashUri }} style={styles.loader} />;

export default App;
