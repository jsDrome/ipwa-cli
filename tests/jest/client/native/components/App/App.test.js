import React from 'react';
import App from '@src/client/native/components/App/App';
import renderer from 'react-test-renderer';

jest.mock('expo-ads-admob', () => ({ AdMobBanner : null }));

test('App -> render', () => {
  const component = renderer.create(
    <App />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
