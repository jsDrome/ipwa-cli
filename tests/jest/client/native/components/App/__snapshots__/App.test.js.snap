// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App -> render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000000",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      bounce={false}
      cacheEnabled={true}
      domStorageEnabled={true}
      javaScriptEnabled={true}
      messagingEnabled={false}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      scrollEnabled={true}
      source={
        Object {
          "uri": "https://ipwa-app.web.app?app=true",
        }
      }
      startInLoadingState={true}
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Object {
            "backgroundColor": "#000000",
            "height": "100%",
            "marginTop": 30,
          },
        ]
      }
      useSharedProcessPool={true}
      useWebKit={true}
      userAgent="Mozilla/5.0 (Linux; Android 7.0; SM-G930V Build/NRD90M) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.125 Mobile Safari/537.36"
    />
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "backgroundColor": "#000000",
          "height": "100%",
          "marginTop": -30,
          "width": "100%",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://ipwa-app.web.app/img/splash.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
